pkg/util/steps/runner.go

```diff
@@ -20,18 +21,42 @@ func friendlyName(f interface{}) string {
 // Step is the interface for steps that Runner can execute.
 type Step interface {
 	run(ctx context.Context, log *logrus.Entry) error
+	setPollInterval(time.Duration)
+	setTimeout(time.Duration)
 	String() string
 }
 
 // Run executes the provided steps in order until one fails or all steps
 // are completed. Errors from failed steps are returned directly.
-func Run(ctx context.Context, log *logrus.Entry, pollInterval time.Duration, steps []Step) error {
+func Run(ctx context.Context, log *logrus.Entry, pollInterval time.Duration, globalTimeout time.Duration, steps []Step, testHook TestHook) error {
```

Comment:
I think we are doing someting wrong if in the production code we have to add an extra param for testing porpuses. I still have a lot to review and I need to dig more into this. Hopefully I'll be able to suggest some alternative way.